src/Agent.Worker/Release/Artifacts/FileShareArtifact.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Agent.Worker.Release.Artifacts.Definition;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts
+{
+    // TODO: Implement serviceLocator pattern to have custom attribute as we have different type of artifacts
+    [ServiceLocator(Default = typeof(FileShareArtifact))]
+    public interface IFileShareArtifact : IAgentService
+    {
+        Task Download(
+            ArtifactDefinition artifactDefinition,
+            IExecutionContext executionContext,
+            string localFolderPath);
+    }
+
+    // TODO: Add test for this
+    public class FileShareArtifact : AgentService, IFileShareArtifact
+    {
+        public async Task Download(ArtifactDefinition artifactDefinition, IExecutionContext executionContext, string localFolderPath)
+        {
+            ArgUtil.NotNull(artifactDefinition, nameof(artifactDefinition));
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNullOrEmpty(localFolderPath, nameof(localFolderPath));
+
+            var dropLocation = GetFileShareDropLocation(artifactDefinition.Details.RelativePath, artifactDefinition.Version);
+            await DownloadArtifact(artifactDefinition, executionContext, dropLocation, localFolderPath);
+        }
+
+        public async Task DownloadArtifact(ArtifactDefinition artifactDefinition, IExecutionContext executionContext, string dropLocation, string localFolderPath)
+        {
+            ArgUtil.NotNull(artifactDefinition, nameof(artifactDefinition));
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNullOrEmpty(localFolderPath, nameof(localFolderPath));
+            ArgUtil.NotNullOrEmpty(dropLocation, nameof(dropLocation));
+
+            var trimChars = new[] { '\\', '/' };
```

Comment:
Here we are trimming download both type of slashed at the end.

Also this FileShare artifact will not work in linux (even in current xplat agent if we use build drop as fileshare it wont work). So far no one has asked us for accessing FileShareBuildDrop from a xplat agent using RM extension. If required we could try mounting it and see if its accessible.